dcrpm/pidutil.py

```diff
@@ -47,6 +68,43 @@ def pids_holding_file(path):
     return procs
 
 
+def pids_holding_file_lsof(lsof, path):
+    # type: (str, str) -> Set[int]
+    try:
+        cmd = "{} -F p {}".format(lsof, path)
+        proc = run_with_timeout(cmd, LSOF_TIMEOUT, raise_on_nonzero=False)
+    except DcRPMException:
+        logger.warning("lsof timed out")
+        return set()
+
+    if proc.returncode != StatusCode.SUCCESS and proc.stderr:
+        # Returning nonzero means either nothing matched or something went
+        # wrong. If nothing matches stderr will be empty; if it contains output
+        # then something went wrong.
+        logger.warning("lsof returned non-zero: %s", proc.stderr)
+
+    pids = set()
+    for line in proc.stdout.splitlines():
+        if line.startswith("p"):
+            pids.add(int(line[1:]))
+    return pids
+
+
+def procs_holding_file_lsof(path):
```

Comment:
If we decide to remove psutil lookup, then this should probably be renamed `procs_holding_file`.